### PR TITLE
First wave of namespace changes for IoT device update

### PR DESCRIPTION
**Pull request by digimaun**
Change root namespace from Azure.Iot -> Azure.IoT to fulfill POR.

### CODE REVIEW

File: sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/GlobalSuppressions.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// a specific target and scoped to a namespace, type, member, etc.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "AZC0001:Use one of the following pre-approved namespace groups (https://azure.github.io/azure-sdk/registered_namespaces.html): Azure.AI, Azure.Analytics, Azure.Communication, Azure.Data, Azure.DigitalTwins, Azure.Iot, Azure.Learn, Azure.Media, Azure.Management, Azure.Messaging, Azure.Search, Azure.Security, Azure.Storage, Azure.Template, Azure.Identity, Microsoft.Extensions.Azure", Justification = "<Pending>", Scope = "namespace", Target = "~N:Azure.IoT.DeviceUpdate")]
+[assembly: SuppressMessage("Usage", "AZC0001:Use one of the following pre-approved namespace groups (https://azure.github.io/azure-sdk/registered_namespaces.html): Azure.AI, Azure.Analytics, Azure.Communication, Azure.Data, Azure.DigitalTwins, Azure.Iot, Azure.Learn, Azure.Media, Azure.Management, Azure.Messaging, Azure.Search, Azure.Security, Azure.Storage, Azure.Template, Azure.Identity, Microsoft.Extensions.Azure", Justification = "<Pending>", Scope = "namespace", Target = "~N:Azure.IoT.DeviceUpdate.Models")]
```
Comment (AlexGhiondea) [inline sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/GlobalSuppressions.cs:12]: Is this still required?

**Review by AlexGhiondea — APPROVED** [review]
Left a comment about a suppression message, but LGTM otherwise.

File: sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/Azure.IoT.DeviceUpdate.csproj [hunk 2]
```diff
@@ -9,6 +9,8 @@
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
     <!-- These supressions should be removed in a production library -->
     <NoWarn>$(NoWarn);CS1591;AZC0001;AZC0012</NoWarn>
+    <AssemblyName>Azure.IoT.DeviceUpdate</AssemblyName>
```
Comment (weshaggard) [inline sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/Azure.IoT.DeviceUpdate.csproj:12]: I believe these are redundant as they default from the project which you renamed it these shouldn't be needed.

File: sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/GlobalSuppressions.cs (hunk 1 above)
Comment (digimaun) [reply sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/GlobalSuppressions.cs:12]: It is just a point-in-time temporary artifact to get through CI which we will remove after we merge https://github.com/Azure/azure-sdk-tools/pull/1492.

File: sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/Azure.IoT.DeviceUpdate.csproj (hunk 2 above)
Comment (digimaun) [reply sdk/deviceupdate/Azure.Iot.DeviceUpdate/src/Azure.IoT.DeviceUpdate.csproj]: Sounds good removed.

**Pull request merged**